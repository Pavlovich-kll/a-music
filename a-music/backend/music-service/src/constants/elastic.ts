export const ELASTIC_URL = 'http://elasticsearch:9200';