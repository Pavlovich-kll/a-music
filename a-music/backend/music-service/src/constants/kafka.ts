export const KAFKA_URL = 'kafka:29092';
