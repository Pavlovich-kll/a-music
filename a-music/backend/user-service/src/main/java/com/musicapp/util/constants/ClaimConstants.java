package com.musicapp.util.constants;

import lombok.experimental.UtilityClass;

/**
 * Константы параметров jwt токена.
 *
 * @author evgeniycheban
 */
@UtilityClass
public class ClaimConstants {

    /**
     * Идентификатор пользователя.
     */
    public final String ID = "id";

    /**
     * Имя пользователя.
     */
    public final String USERNAME = "username";

    /**
     * Роли пользователя.
     */
    public final String ROLES = "roles";

    /**
     * Верифицированный номер телефона пользователя.
     */
    public final String VERIFIED_PHONE = "verifiedPhone";

    /**
     * Верифицироанный e-mail
     */
    public final String VERIFIED_EMAIL = "verifiedEmail";
}
