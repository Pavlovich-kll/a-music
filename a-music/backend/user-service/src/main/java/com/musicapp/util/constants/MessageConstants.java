package com.musicapp.util.constants;

import lombok.experimental.UtilityClass;

/**
 * Константы сообщений об ошибках.
 *
 * @author a.nagovicyn
 */
@UtilityClass
public class MessageConstants {

    /**
     * Сообщение об отсутствии логина пользователя при регистрации
     */
    public static final String USER_LOGIN_EMPTY = "oauth2.user.login.empty";

    /**
     * Сообщение о занятом логине пользователя при регистрации
     */
    public static final String USER_LOGIN_EXISTS = "oauth2.user.alreadyExists";

    /**
     * Сообщение о том, что введенный и повторенный пароль не совпадают
     */
    public static final String USER_REPEAT_PASSWORD_NOT_MATCH = "user.repeatPassword.ne";

    /**
     * Сообщение об отсутствующем в базе email коде
     */
    public static final String EMAIL_CODE_INCORRECT = "emailCode.code.incorrect";

    /**
     * Сообщение об отсутствии заявки с таким статусом
     */
    public static final String INVITE_STATUS_NOT_FOUND = "friendInvite.status.notFound";

    /**
     * Сообщение об отсутствии заявки в друзья пользователю
     */
    public static final String INVITE_ACCEPTOR_NOT_FOUND = "friendInvite.acceptor.notFound";

    /**
     * Сообщение об отсутствии заявки в друзья от пользователю
     */
    public static final String INVITE_INITIATOR_NOT_FOUND = "friendInvite.initiator.notFound";

    /**
     * Сообщение об отсуствии пользователя с таким id
     */
    public static final String USER_ID_NOT_FOUND = "user.id.notFound";

    /**
     * Сообщение об отсуствии концерта с таким id
     */
    public static final String CONCERT_ID_NOT_FOUND = "concert.id.notFound";

    /**
     * Сообщение об отсутствии билета с таким id
     */
    public static final String TICKET_ID_NOT_FOUND = "ticket.id.notFound";

    /**
     * Сообщение о неправильном формате загружаемого файла
     */
    public static final String FILE_TYPE_NOT_SUPPORTED = "image.format.notSupported";

    /**
     * Сообщение об отсуствии города с таким id
     */
    public static final String CITY_ID_NOT_FOUND = "city.id.notFound";

    /**
     * Сообщение об отсуствии страны с таким id
     */
    public static final String COUNTRY_ID_NOT_FOUND = "country.id.notFound";

    /**
     * Сообщение об отсуствии файла с таким именем
     */
    public static final String FILE_NAME_NOT_FOUND = "file.name.notFound";
    /**
     * Сообщение об отсуствии плейлиста с таким id
     */
    public static final String PLAYLIST_ID_NOT_FOUND = "playlist.id.notFound";

    /**
     * Сообщение об отсуствии подписки с таким id
     */
    public static final String SUBSCRIPTION_ID_NOT_FOUND = "subscription.id.notFound";

    /**
     * Сообщение указывает на отсутствие значения в объекте UserSubscriptionEmailCode
     */
    public static final String USER_SUBSCRIPTION_EMAIL_CODE_NOT_FOUND = "userSubscriptionEmailCode.notFound";

    /**
     * Сообщение об отсуствии аудио с таким id
     */
    public static final String AUDIO_ID_NOT_FOUND = "audio.id.notFound";
}
