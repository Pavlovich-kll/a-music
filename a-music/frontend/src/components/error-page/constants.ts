export const NOT_FOUND = '404';
export const SERVER_ERROR = '505';

export const NOT_FOUND_DESC = 'The page you requested has not been found.';
export const SERVER_ERROR_DESC = 'Internal server error.';
