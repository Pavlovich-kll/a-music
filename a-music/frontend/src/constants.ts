export const USER_ACTION: string = 'USER/COMMIT_ACTION';
export const CONTENT_ACTION_TYPE: string = 'CONTENT/COMMIT_ACTION';
export const FRIENDS_ACTION_TYPE: string = 'FRIENDS/COMMIT_ACTION';

export const GET_PLAY_LIST_FILE: string = `${process.env.API_URL}music-service/content/file`;
export const BASE_URL: string = `${process.env.API_URL}music-service/file`;

// Избавиться когда сделают friends
export const FAKE_IMG: string =
  'https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Mona_Lisa%2C_by_Leonardo_da_Vinci%2C_from_C2RMF_retouched.jpg/250px-Mona_Lisa%2C_by_Leonardo_da_Vinci%2C_from_C2RMF_retouched.jpg';
