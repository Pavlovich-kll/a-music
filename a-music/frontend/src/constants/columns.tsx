import { Link } from 'react-router-dom';
import React from 'react';

export const columns = [
  {
    title: 'Title',
    dataIndex: 'title',
    key: 'title',
    render: (text: any, item: any) => <Link to={`/collection/${item._id}`}>{text}</Link>,
    width: 300,
    ellipsis: true,
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    ellipsis: true,
  },
  { title: 'Tracks count', dataIndex: 'track_count', key: 'track_count', width: 130 },
];
