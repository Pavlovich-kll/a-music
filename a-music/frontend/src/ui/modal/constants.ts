export const MODAL_SIZE = {
  small: '30%',
  medium: '50%',
  large: '60%',
  authSize: '70%',
};

export type SizeType = keyof typeof MODAL_SIZE;
