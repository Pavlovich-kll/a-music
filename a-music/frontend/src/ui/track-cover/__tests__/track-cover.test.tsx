import React from 'react';
import TrackCover from '../index';
import { shallow } from 'enzyme';
import { BASE_URL } from '../../../constants';

let wrapper: any;

describe('TrackCover', () => {
  test(`should render`, () => {
    wrapper = shallow(<TrackCover />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should display default values when author and title are not passed', () => {
    wrapper = shallow(<TrackCover />);
    expect(wrapper.find('.track').text()).toEqual('title - author');
  });

  test('should display title and author when they are passed', () => {
    wrapper = shallow(<TrackCover title="Poison" author="Alice Cooper" />);
    expect(wrapper.find('.track').text()).toEqual('Poison - Alice Cooper');
  });

  test('should display alt when img is not displayed', () => {
    wrapper = shallow(<TrackCover title="Poison" author="Alice Cooper" cover_id="0assda" />);
    expect(wrapper.find('img').prop('alt')).toBe('Poison');
  });

  test('src should have exact structure', () => {
    wrapper = shallow(<TrackCover cover_id="12mdkasdk2" />);
    expect(wrapper.find('img').prop('src')).toBe(`${BASE_URL}/12mdkasdk2`);
  });
});
