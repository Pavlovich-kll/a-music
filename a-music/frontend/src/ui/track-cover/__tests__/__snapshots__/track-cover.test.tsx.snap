// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackCover should render 1`] = `
<styled.div>
  <span
    className="track"
  >
    title
     - 
    author
  </span>
</styled.div>
`;
